src/librustdoc/html/url_parts_builder.rs

```diff
@@ -105,10 +105,18 @@ impl UrlPartsBuilder {
 
 /// This is just a guess at the average length of a URL part,
 /// used for [`String::with_capacity`] calls in the [`FromIterator`]
-/// and [`Extend`] impls.
+/// and [`Extend`] impls, and for [estimating item path lengths].
 ///
-/// This is intentionally on the lower end to avoid overallocating.
-const AVG_PART_LENGTH: usize = 5;
+/// The value `8` was chosen for a few reasons:
+///
+/// * It seems like a good guess for the average part length.
+/// * It's a power of two, which is generally good when allocating memory.
+/// * It's one of jemalloc's size classes, so it's especially good when
+///   allocating memory. See [this comment] for more.
```

Comment:
It's not quite right... I suggest replacing the last two bullet points with this:
```
/// * jemalloc's size classes are all multiples of eight, which means that the amount of memory
///   it allocates will often match the amount requested, avoiding wasted bytes.
```

Reply:
Done, thanks!